### DISCUSSION TITLE
Custom formatter deletes entire file if command fails.

### DISCUSSION THREAD

**Discussion by aspizu** [helix-editor/helix#8427] [Troubleshooting]
```sh
#!/bin/bash
set -e
isort - | ruff check --fix - | ruff format -
```

```toml
formatter = { command = "/home/aspizu/.config/helix/python-format.sh", args = [] }
```

This will delete entire file if the command fails.

**Comment by kirawi** [comment]
The formatter needs to return a non-zero exit status in order for it to be treated as an error.

**Reply by aspizu** [reply]
It does return a non-zero exit status (`2`).

**Reply by kirawi** [reply]
Can you share the file where this happens?

**Reply by aspizu** [reply]
My bash script seems to always return `0`.

**Comment by aspizu** [comment]
Had to use
```bash
set -o pipefail
```
